go-controller/pkg/ovn/gateway_init.go

```diff
@@ -263,10 +263,29 @@ func (oc *DefaultNetworkController) gatewayInit(nodeName string, clusterIPSubnet
 	externalRouterPort := types.GWRouterToExtSwitchPrefix + gatewayRouter
 
 	nextHops := l3GatewayConfig.NextHops
-	if len(l3GatewayConfig.NextHops) == 0 {
-		nextHops = node.DummyNextHopIPs()
-		if err := gateway.CreateDummyGWMacBinding(oc.sbClient, nodeName); err != nil {
-			return err
+
+	if err := gateway.CreateDummyGWMacBindings(oc.sbClient, nodeName); err != nil {
+		return err
+	}
+
+	for _, nextHop := range node.DummyNextHopIPs() {
+		// Add return service route for OVN back to host
+		prefix := types.V4MasqueradeSubnet
+		if utilnet.IsIPv6(nextHop) {
+			prefix = types.V6MasqueradeSubnet
+		}
+		lrsr := nbdb.LogicalRouterStaticRoute{
+			IPPrefix:   prefix,
+			Nexthop:    nextHop.String(),
+			OutputPort: &externalRouterPort,
+		}
+		p := func(item *nbdb.LogicalRouterStaticRoute) bool {
+			return item.OutputPort != nil && *item.OutputPort == *lrsr.OutputPort && item.IPPrefix == lrsr.IPPrefix
```

Comment:
don't u need to make sure `iterm.NextHop == lrsr.NetHop` ?

Reply:
take a look at https://github.com/ovn-org/ovn-kubernetes/pull/3359

I'm trying to fix that behavior everywhere in the code so that we use unique routes. Therefore the nexthop should not be in the predicate because we should update the nexthop if we find the route already exists matching the prefix.